test/hold_effect_gems.c

```diff
@@ -0,0 +1,87 @@
+#include "global.h"
+#include "test_battle.h"
+
+ASSUMPTIONS
+{
+    ASSUME(gItems[ITEM_NORMAL_GEM].holdEffect == HOLD_EFFECT_GEMS);
+}
+
+SINGLE_BATTLE_TEST("Gem is consumed when it corresponds to the type of a move")
+{
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET) { Item(ITEM_NORMAL_GEM); };
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(player, MOVE_EMBER); }
+        TURN { MOVE(player, MOVE_TACKLE); }
+    } SCENE {
+        NONE_OF {
+            ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_HELD_ITEM_EFFECT, player);
+            MESSAGE("Fire Gem strengthened Wobbuffet's power!");
+        }
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_EMBER, player);
+        ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_HELD_ITEM_EFFECT, player);
+        MESSAGE("Normal Gem strengthened Wobbuffet's power!");
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_TACKLE, player);
+    }
+}
+
+SINGLE_BATTLE_TEST("Gem boost is only applied once", s16 damage)
+{
+    u16 item;
+
+    PARAMETRIZE { item = ITEM_NONE; }
+    PARAMETRIZE { item = ITEM_NORMAL_GEM; }
+    PARAMETRIZE { item = ITEM_NONE; }
```

Comment:
I want to get the normal damage without any boosts, then gem boosted damage and a 3rd time to verify that the gem boost is only applied once.

Now that I think about it the first hit might be redundant. It would probably be enough to compare the second and third hit.

Reply:
I think if you want to check that it's not applied a second time, you probably want two `TURN`s, rather than `PARAMETRIZE` like in your first test? Because with `PARAMETRIZE` the three battles are completely separate.

Reply:
Oh yeah, then this test makes no sense. lol

Will fix it, thanks.

Reply:
Fixed.